biogrid/libs/biogrid-simulator/src/lib/biobrain/biobrain-class.ts

```diff
@@ -375,7 +371,20 @@ export class BioBrain implements Brain {
             totalResistance
           );
           // Get the total energy which can be supplied by the supplying agent
-          const energyInSupplier = await supplyingAgents[index].getEnergyInJoules();
+          let energyInSupplier: Energy;
+          if (
+            supplyingAgents[index].gridItemName.includes(
+              GRID_ITEM_NAMES.SOLAR_PANEL
+            )
+          ) {
+            energyInSupplier = await (supplyingAgents[
+              index
+            ] as SolarPanel).getEnergyInJoules(date);
+          } else {
+            energyInSupplier = await Promise.resolve(
+              supplyingAgents[index].getEnergyInJoules()
+            );
```

Comment:
is this supposed to be a part of this change?

Reply:
Yeah, it calls the weather API and thus we wait for results.

Reply:
I'm just wondering why this change is part of this particular PR - this PR seems to be writing JS docs, so maybe this should be split out into a separate PR